app.py

```
@@ -1725,6 +1736,20 @@ def handle_activity_response(room_name, user_response, username):
 
             # Check if the step has a question
             if "question" in step:
+                # Execute pre-script if it exists (runs before categorization, with user_response available)
+                if "pre_script" in step:
+                    print(f"DEBUG: Executing pre-script")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)